pwiz_tools/Skyline/Model/Results/SpectraChromDataProvider.cs

```diff
@@ -292,10 +294,13 @@ private void ExtractChromatogramsLocked()
                 }
 
                 UpdatePercentComplete();
-

```

Comment:
Is this CurrentSpectrum call slow and that's why it was in the if blocks below? If so, it could be changed to only be called when _globalSpectrumClassFilter != null.